### PR TITLE
add new gradlew commands

### PR DESCRIPTION
**Pull request by thomas-k-cameron**
update readme to include new gradlew commands

## Motivation and Context
New you can now test/check with cargo --all-features

## Description
adds gradlew command that runs cargo test/check with --all-features

## Testing
NA

I wasn't quite sure how to test these stuff

## Checklist
- [ ] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [ ] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by thomas-k-cameron** [comment]
@Velfi 
Hi,

Concerning the new commands, I made a separate pull request

**Review by Velfi — APPROVED** [review]

**Comment by Velfi** [comment]
These look good to me, no test necessary. I'll merge it once CI passes.

**Comment by Velfi** [comment]
Looks like I spoke too soon 😓 

You should get the same errors as CI when you run things locally. Are you seeing that?

**Comment by thomas-k-cameron** [comment]
fixed it up.
Seems like I messed it up when I switched to a different branch.

**Comment by thomas-k-cameron** [comment]
I pushed a change.
Can you see if it work?

**Comment by thomas-k-cameron** [comment]
Ok, apparently I checked if it is going to work on a different branch and thought I'm ready.

Let me fix it up.

**Comment by thomas-k-cameron** [comment]
@Velfi 
I think it's working now. Can you check it out?

**Comment by thomas-k-cameron** [comment]
It appears that your PR bot didn't work, but judging from the error message, I don't think there is something that I can do it on my side.

**Comment by jdisanti** [comment]
I'm more inclined to update the existing `cargoTest` and `cargoCheck` targets to include `--all-features`. To me, it's a bug that they don't since CI does.

> It appears that your PR bot didn't work, but judging from the error message, I don't think there is something that I can do it on my side.

Yeah, this is fine.

**Comment by thomas-k-cameron** [comment]
@jdisanti 
Fixed

**Review by jdisanti — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ Project Layout
   Common commands:
   * `./gradlew :aws:sdk:assemble`: Generate (but do not test / compile etc.) a fresh SDK into `sdk/build/aws-sdk`
   * `./gradlew :aws:sdk:test`: Generate & run all tests for a fresh SDK
-  * `./gradlew :aws:sdk:{cargoCheck, cargoTest, cargoDocs, cargoClippy}`: Generate & run specified cargo command.
+  * `./gradlew :aws:sdk:{cargoCheck, cargoTest, cargoDocs, cargoClippy, cargoCheckAllFeatures, cargoTestAllFeatures}`: Generate & run specified cargo command.
```
Comment (jdisanti) [inline README.md:34]:
```suggestion
  * `./gradlew :aws:sdk:{cargoCheck, cargoTest, cargoDocs, cargoClippy}`: Generate & run specified cargo command.
```

**Comment by jdisanti** [comment]
Thank you for the contribution! I'll get this merged.

File: buildSrc/src/main/kotlin/CodegenTestCommon.kt [hunk 2]
```diff
@@ -233,16 +233,17 @@ fun Project.registerCargoCommandsTasks(
             "generateCargoConfigToml",
             this.tasks.findByName("modifyMtime")?.let { "modifyMtime" },
         )
+
     this.tasks.register<Exec>(Cargo.CHECK.toString) {
         dependsOn(dependentTasks)
         workingDir(outputDir)
-        commandLine("cargo", "check", "--lib", "--tests", "--benches")
+        commandLine("cargo", "check", "--lib", "--tests", "--benches", "-all--features")
```
Comment (jdisanti) [inline buildSrc/src/main/kotlin/CodegenTestCommon.kt:240]:
```suggestion
        commandLine("cargo", "check", "--lib", "--tests", "--benches", "--all-features")
```

File: buildSrc/src/main/kotlin/CodegenTestCommon.kt [hunk 3]
```diff
@@ -233,16 +233,17 @@ fun Project.registerCargoCommandsTasks(
...
     }
 
     this.tasks.register<Exec>(Cargo.TEST.toString) {
         dependsOn(dependentTasks)
         workingDir(outputDir)
-        commandLine("cargo", "test")
+        commandLine("cargo", "test", "-all--features")
```
Comment (jdisanti) [inline buildSrc/src/main/kotlin/CodegenTestCommon.kt:246]:
```suggestion
        commandLine("cargo", "test", "--all-features")
```

**Pull request merged**